packages/gatsby-theme-styleguide/src/utils/copy-to-clipboard.js

```diff
@@ -0,0 +1,29 @@
+const copyToClipboard = str => {
```

Comment:
Test here if the browser is compatible with the copy API your components to manage state. 
You can catch errors and reject to let the component make their own error handling. 

inside your component the usage should look something like : 

```js
copyToClipBoard(somethingToCopy)
  .then(() => setSate(...))
  .catch(e => handleError(...))
```

we could ba able to remove a lot of duplication in the component using it

Reply:
Added the catch to the function